### ISSUE TITLE
✨ [REQUEST] Tandoor-HA Share dir

### ISSUE THREAD

**Issue by jimmyz5150** [alexbelgium/hassio-addons#865]
### Checked

- [X] If a new addon, I have checked on  Google that such as addon doesn't already exists
- [X] If a new feature, I have searched in the discussions & issues that it was not already covered

### Which addon?

Home Assistant Tandoor

### Is your feature request related to a problem? Please describe

The latest HA 2023.6 update allows for sharing folders to a NAS. Add functionality to map HA directly to shared "Network Storage" 
I'm using an ArgonOne with M.2 SATA SSD for HA and would to use this shared folder for Tandoor Recipes.

### Describe the solution you'd like

Now that Home Assistant opens up a share folder for NAS.
Enable HA Tandoor config to setup for a local option, map maybe via "External Recipes" to the share folder inside of HA
(192.168.x.xxx  \share\'your recipes folder').  

I have a shared HA network folder via my QNAP that has my recipe PDFs.

Adding this would possibly eliminate the need to postges external db and use the SQLite db option (since this works for me).
OR is this do-able already and don't know the setup?

I think Tandoor integration is my only option for HA since my recipes are PDF's. Grocy is way too needy to setup. I just want to view my recipes.

For some reason I've never been able to get external recipes setup correctly for QNAP/postgres. I have a docker up but can't point it to recipes on QNAP and as I am not the most proficient in these things... this seems to be a better option.

Thanks for considering

**Comment by nough** [comment]
I've added a default feature to the add-on that creates a folder in `/config/addons_config/tandoor_recipes/externalfiles`. This can be mapped within tandoor as `/opt/recipes/externalfiles`

I'd like to set this as a config option but I'm not sure how easy it would be. For now you could create a symlink from your storage location to that folder.

`ln -s /path/to/storage/ /config/addons_config/tandoor_recipes/externalfiles/`

That may work - you want another folder within externalfiles.

**Comment by alexbelgium** [comment]
Good idea, implemented :-) I'll push the version this afternoon after the stats daemon has ran

**Comment by nough** [comment]
That's what I wanted to do when I made my pull request, but didnt understand how to do it. Worth updating the documentation as well, I'll try doing a pull request in future if you don't get around to it today.

**Comment by nough** [comment]
I was thinking it may be better to map in /media/ and put the default externalfiles there instead of the config folder. But again, didn't understand how to do that.

**Comment by alexbelgium** [comment]
As default? We could do that but it would break users installations if they currently use the externalfiles folder in /config

**Comment by nough** [comment]
I was initially thinking this was the better idea but didn't know how to do it when I submitted my pull request. I think since that pull was so recent it shouldn't affect too many people - maybe just me!

**Comment by alexbelgium** [comment]
According to my stats, 240 people downloaded (installation or update) the add-on after your PR :)

**Comment by nough** [comment]
So 240 people had auto-updates turned on and didn't read the changelog?

**Comment by nough** [comment]
But you're right. Is it possible to map in /media/ anyway so that users can set their own media folder?

**Comment by jimmyz5150** [comment]
Hey guys awesome working on this! I just wanted to put my two cents in.. Wouldn't it be better to map to /share/ vs /media/ ? Once the db is setup no one should need to see their recipes folder and would have to use samba share to add files to recipes in HA. The reason I say that is it media is already used by HA and visible in side bar. Share folder would be in the backend of sorts and users would have to open Tandoor to view recipes vs clicking on media in side bar and cluttering up media directory.
Just a thought, thanks

**Comment by alexbelgium** [comment]
Mmh actually both are feasible rather easily. If you map your network storage to /media/ha for example using the HA embedded network feature, you just need to write /media/ha in the externalfiles_folder option and it will get them from there

**Comment by jimmyz5150** [comment]
All right All right.. she's a working via /share... Awesomeness. One thing is my /share/JZsRecipes dir is not allowing or showing the sub directories i.e Southwest or Chinese foods dir. I deleted add-on config/tandor dir and removed Tandoor from HA and re installed just in case. On new install all that was needed was HOST, SECRET KEY( which isn't hidden in logs) and share dir for HA... Any thoughts??

**Comment by nough** [comment]
You need to register them separately. So
/Dir/externalfiles is your external location
But then for each folder in that you need to register a location in the
first tandoor externalfiles menu, e.g. `/dir/externalfiles/Chinese/` and
`/dir/externalfiles/porridge`

Is that the issue? It's the way Tandoor works.

**Comment by jimmyz5150** [comment]
I figured one would end up adding sub dir separately totally fine, happy it's working great job. I figured out books and will probably use that mostly to categorize. Can you add folder dirs to books or do you have to add recipes one by one?

**Comment by nough** [comment]
I was going to suggest trying the Django admin backend but that's even
worse than the front-end. Yes, it's one by one as far as I can tell.

**Comment by jimmyz5150** [comment]
I've actually been using the admin backend some what while I was trying to get Tandoor working. I can multitask better there.

**Comment by nough** [comment]
Ok, I've tested this now. The config option is not needed.

The config option will create the folder you want to use if it doesn't already exist. But as media, config and share are already mapped in, you can just reference them directly in tandoor. With folders you have created manually.

Worth including in documents

**Comment by jimmyz5150** [comment]
hey thanks to all... should I close out?

**Comment by alexbelgium** [comment]
If you are happy with how it is now! I mean in the limits of what we can do ;)

**Comment by jimmyz5150** [comment]
Thanks again for the awesome work!
